action.yml

```diff
@@ -17,13 +17,30 @@ runs:
     # * https://github.com/JuliaLang/Pkg.jl/issues/2011
     # * https://github.com/JuliaRegistries/General/issues/16777
     # * https://github.com/JuliaPackaging/PkgServer.jl/issues/60
-  - run: julia --color=yes -e 'using Pkg; VERSION >= v"1.5-" && !isdir(joinpath(DEPOT_PATH[1], "registries", "General")) && Pkg.Registry.add("General")'
+  - run: |
+      julia --color=yes -e '
+        using Pkg
+        if VERSION >= v"1.5-" && !isdir(joinpath(DEPOT_PATH[1], "registries", "General"))
+          function f()
+              @info("Attempting to clone the General registry")
+              Pkg.Registry.add("General")
+          end
+          delays = ExponentialBackOff(; n = 10, max_delay = 120)
+          try
+              retry(f; delays = delays)()
+              @info("Successfully cloned the General registry")
+          catch ex
+              msg = "I was unable to clone the General registry. However, the build will continue."
+              @error(msg, exception=(ex,catch_backtrace()))
+          end
+        end'
```

Comment:
This is long enough that we might want to move it to a separate file and call it with `julia --color=yes add_registry.jl` instead. Makes it easier to spot mistakes because syntax highlighting etc works.

But that can also be done in a follow-up PR, I don't mind.

Reply:
Done